modules/instruments/php/module.class.inc

```diff
@@ -59,15 +59,21 @@ class Module extends \Module
         $params    = $request->getQueryParams();
         $commentID = $params['commentID'] ?? null;
         if (empty($commentID)) {
-            return (new \Laminas\Diactoros\Response())
-                ->withBody(new \LORIS\Http\StringStream("Missing CommentID"))
-                ->withStatus(400);
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                "Missing CommentID"
+            );
+        }
+        try {
+            $instrument = \NDB_BVL_Instrument::factory(
+                $instrumentName,
+                $commentID,
+                $page
+            );
```

Comment:
Can you move the 2 lines

```
$request = $request->withAttribute('pageclass', $instrument);
return $instrument->process($request, $instrument);
```

into the try block so that the `$instrument` variable is unambiguously in scope?

Reply:
Done